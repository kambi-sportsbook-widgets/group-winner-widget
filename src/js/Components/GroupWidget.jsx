import React, { Component, PropTypes } from 'react';
import { widgetModule } from 'kambi-widget-core-library';
import { IconHeader, ScrolledList, TabPagination } from 'kambi-widget-components';
import isMobile from '../Services/mobile';
import BlendedImage from './BlendedImage';
import CountryList from './CountryList/CountryList';
import CountryListElement from './CountryList/CountryListElement/CountryListElement';
import styles from './GroupWidget.scss';

/**
 * Navigates to given group (event) page.
 * @param {object} group Tournament group (event entity)
 */
const onGroupClick = function(group) {
   if (group.event.openForLiveBetting === true) {
      widgetModule.navigateToLiveEvent(group.event.id);
   } else {
      widgetModule.navigateToEvent(group.event.id);
   }
};

class GroupWidget extends Component {

   /**
    * Constructs.
    * @param {object} props Component properties
    */
   constructor(props) {
      super(props);
      widgetModule.enableWidgetTransition(true);
   }

   /**
    * Called after component mounts
    */
   componentDidMount() {
      widgetModule.adaptWidgetHeight();
   }

   /**
    * Called after updating component's DOM.
    */
   componentDidUpdate() {
      widgetModule.adaptWidgetHeight();
   }

   /**
    * Renders widget.
    * @returns {XML}
    */
   render() {
      const renderTab = idx => <div key={idx} className={styles.tab}>{this.props.groups[idx].groupName}</div>;

      return (
         <div>
            <IconHeader title={this.props.title} subtitle={this.props.tagline}>
               <BlendedImage imgPath='src/img/football.svg' />
            </IconHeader>
            <TabPagination
               renderTab={renderTab}
               selected={this.props.selected}
               renderTabList={args => <ScrolledList {...args} showControls={!isMobile()} />}
            >
               {this.props.groups.map(group => (
                  <CountryList key={group.event.id}>
                     {group.betOffers[0].outcomes.map(outcome => (
                        <CountryListElement
                           key={outcome.id}
                           flagUrl={`${this.props.flagUrl}${outcome.participantId}.svg`}
                           country={outcome.label}
                           outcome={outcome}
                           onClick={onGroupClick.bind(null, group)}
                        />
                     ))}
                  </CountryList>
               ))}
            </TabPagination>
         </div>
      );
   }
}

GroupWidget.defaultProps = {
   selected: 0
};

GroupWidget.propTypes = {

   /**
    * Tournament groups array (event entities)
    */
   groups: PropTypes.array.isRequired,

   /**
    * Widget's title. Will be figured out if not provided explicitly.
    */
   title: PropTypes.string,

   /**
    * Widget's tag line. Will be figured out if not provided explicitly.
    */
   tagline: PropTypes.string,

   /**
    * Base URL of country flags
    */
   flagUrl: PropTypes.string,

   /**
    * Selected group index (default to 0)
    */
   selected: PropTypes.number

};

export default GroupWidget;
