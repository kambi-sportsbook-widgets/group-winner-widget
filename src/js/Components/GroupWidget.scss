:local(.tab) {
   margin: 0 auto;
}

:local(.button) {
   background-color:red;
   display: flex;
   justify-content: center;
}
