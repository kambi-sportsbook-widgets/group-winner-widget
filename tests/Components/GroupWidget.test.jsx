/* eslint-env jest */
import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import { mount, shallow } from 'enzyme'
import GroupWidget from '../../src/js/Components/GroupWidget'
import { widgetModule } from 'kambi-widget-core-library'

let renderer

describe('', () => {
  it('', () => {})
})

// jest.mock('kambi-widget-core-library', () => ({
//    widgetModule: {
//       navigateToEvent: jest.fn(),
//       navigateToLiveEvent: jest.fn(),
//       enableWidgetTransition: jest.fn(),
//       adaptWidgetHeight: jest.fn(),
//       betslipIds: []
//    },
//    coreLibrary: {
//       config: {
//          oddsFormat: 'decimal'
//       }
//    },
//    utilModule: {
//       getOddsDecimalValue: _ => _.toFixed(1),
//       getOutcomeLabel: () => 'Outcome label from util'
//    },
//    eventsModule: {
//       subscribe: jest.fn(),
//       unsubscribe: jest.fn()
//    }
// }));

// const mockGroups = [
//    {
//       groupName: 'Test group name',
//       event: { id: 1000 },
//       betOffers: [{
//          outcomes: [
//             {
//                id: 1100,
//                participantId: 1110,
//                label: 'Test outcome label'
//             },
//             {
//                id: 1200,
//                participantId: 1220,
//                label: 'Test outcome label 2'
//             }
//          ]
//       }]
//    },
//    {
//       groupName: 'Test group name',
//       event: {
//          id: 2000,
//          openForLiveBetting: true
//       },
//       betOffers: [{
//          outcomes: [
//             {
//                id: 2100,
//                participantId: 2110,
//                label: 'Test outcome label'
//             },
//             {
//                id: 2200,
//                participantId: 2220,
//                label: 'Test outcome label 2'
//             }
//          ]
//       }]
//    }
// ];

// jest.useFakeTimers();

// describe('GroupWidget.render()', () => {

//    beforeEach(() => {
//       renderer = new ReactShallowRenderer();
//    });

//    it('renders correctly with default props', () => {
//       expect(renderer.render(
//          <GroupWidget
//             groups={[]}
//             title="Test title"
//             tagline="Test subtitle"
//             flagUrl="/base/flags/path/"
//          />
//       )).toMatchSnapshot();
//    });

//    it('renders correctly with groups', () => {
//       expect(renderer.render(
//          <GroupWidget
//             groups={mockGroups}
//             title="Test title"
//             tagline="Test subtitle"
//             flagUrl="/base/flags/path/"
//          />
//       )).toMatchSnapshot();
//    });

//    it('renders correctly with initially selected group', () => {
//       expect(renderer.render(
//          <GroupWidget
//             groups={mockGroups}
//             title="Test title"
//             tagline="Test subtitle"
//             flagUrl="/base/flags/path/"
//             selected={1}
//          />
//       )).toMatchSnapshot();
//    });

// });

// describe('GroupWidget clicks', () => {

//    beforeAll(() => {
//       widgetModule.navigateToEvent.mockClear();
//       widgetModule.navigateToLiveEvent.mockClear();
//    });

//    it('handles click on regular event correctly', () => {
//       const wrapper = mount(
//          <GroupWidget
//             groups={mockGroups}
//             title="Test title"
//             tagline="Test subtitle"
//             flagUrl="/base/flags/path/"
//          />
//       );

//       expect(widgetModule.navigateToEvent).not.toHaveBeenCalled();
//       wrapper.find('.row .flag').first().simulate('click');
//       expect(widgetModule.navigateToEvent).toHaveBeenCalledTimes(1);
//    });

//    it('handles click on live event correctly', () => {
//       const wrapper = mount(
//          <GroupWidget
//             groups={mockGroups}
//             title="Test title"
//             tagline="Test subtitle"
//             flagUrl="/base/flags/path/"
//             selected={1} // <- HERE
//          />
//       );

//       expect(widgetModule.navigateToLiveEvent).not.toHaveBeenCalled();
//       wrapper.find('.row .flag').first().simulate('click');
//       expect(widgetModule.navigateToLiveEvent).toHaveBeenCalledTimes(1);
//    });

// });

// describe('GroupWidget height and transition', () => {

//    beforeEach(() => {
//       widgetModule.adaptWidgetHeight.mockClear();
//       widgetModule.enableWidgetTransition.mockClear();
//    });

//    it('initializes correctly', () => {
//       expect(widgetModule.enableWidgetTransition).not.toHaveBeenCalled();
//       expect(widgetModule.adaptWidgetHeight).not.toHaveBeenCalled();

//       const wrapper = mount(
//          <GroupWidget
//             groups={mockGroups}
//             title="Test title"
//             tagline="Test subtitle"
//             flagUrl="/base/flags/path/"
//          />
//       );

//       expect(widgetModule.enableWidgetTransition).toHaveBeenCalledTimes(1);
//       expect(widgetModule.adaptWidgetHeight).toHaveBeenCalledTimes(1);
//    });

//    it('updates correctly', () => {
//       expect(widgetModule.adaptWidgetHeight).not.toHaveBeenCalled();

//       const wrapper = mount(
//          <GroupWidget
//             groups={mockGroups}
//             title="Test title"
//             tagline="Test subtitle"
//             flagUrl="/base/flags/path/"
//          />
//       );

//       expect(widgetModule.adaptWidgetHeight).toHaveBeenCalledTimes(1);

//       wrapper.find('ItemContainer').first().simulate('click');

//       jest.runOnlyPendingTimers();

//       expect(widgetModule.adaptWidgetHeight).toHaveBeenCalledTimes(2);
//    });

// });
