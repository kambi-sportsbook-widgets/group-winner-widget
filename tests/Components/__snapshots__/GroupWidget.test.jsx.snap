// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupWidget.render() renders correctly with default props 1`] = `
<div>
  <IconHeader
    subtitle="Test subtitle"
    title="Test title"
  >
    <BlendedImage
      imgPath="assets/football.svg"
    />
  </IconHeader>
  <TabPagination
    renderTab={[Function]}
    renderTabList={[Function]}
    selected={0}
  >
    
  </TabPagination>
</div>
`;

exports[`GroupWidget.render() renders correctly with groups 1`] = `
<div>
  <IconHeader
    subtitle="Test subtitle"
    title="Test title"
  >
    <BlendedImage
      imgPath="assets/football.svg"
    />
  </IconHeader>
  <TabPagination
    renderTab={[Function]}
    renderTabList={[Function]}
    selected={0}
  >
    <CountryList>
      <CountryListElement
        country="Test outcome label"
        flagUrl="/base/flags/path/1110.svg"
        onClick={[Function]}
        outcome={
          Object {
            "id": 1100,
            "label": "Test outcome label",
            "participantId": 1110,
          }
        }
      />
      <CountryListElement
        country="Test outcome label 2"
        flagUrl="/base/flags/path/1220.svg"
        onClick={[Function]}
        outcome={
          Object {
            "id": 1200,
            "label": "Test outcome label 2",
            "participantId": 1220,
          }
        }
      />
    </CountryList>
    <CountryList>
      <CountryListElement
        country="Test outcome label"
        flagUrl="/base/flags/path/2110.svg"
        onClick={[Function]}
        outcome={
          Object {
            "id": 2100,
            "label": "Test outcome label",
            "participantId": 2110,
          }
        }
      />
      <CountryListElement
        country="Test outcome label 2"
        flagUrl="/base/flags/path/2220.svg"
        onClick={[Function]}
        outcome={
          Object {
            "id": 2200,
            "label": "Test outcome label 2",
            "participantId": 2220,
          }
        }
      />
    </CountryList>
  </TabPagination>
</div>
`;

exports[`GroupWidget.render() renders correctly with initially selected group 1`] = `
<div>
  <IconHeader
    subtitle="Test subtitle"
    title="Test title"
  >
    <BlendedImage
      imgPath="assets/football.svg"
    />
  </IconHeader>
  <TabPagination
    renderTab={[Function]}
    renderTabList={[Function]}
    selected={1}
  >
    <CountryList>
      <CountryListElement
        country="Test outcome label"
        flagUrl="/base/flags/path/1110.svg"
        onClick={[Function]}
        outcome={
          Object {
            "id": 1100,
            "label": "Test outcome label",
            "participantId": 1110,
          }
        }
      />
      <CountryListElement
        country="Test outcome label 2"
        flagUrl="/base/flags/path/1220.svg"
        onClick={[Function]}
        outcome={
          Object {
            "id": 1200,
            "label": "Test outcome label 2",
            "participantId": 1220,
          }
        }
      />
    </CountryList>
    <CountryList>
      <CountryListElement
        country="Test outcome label"
        flagUrl="/base/flags/path/2110.svg"
        onClick={[Function]}
        outcome={
          Object {
            "id": 2100,
            "label": "Test outcome label",
            "participantId": 2110,
          }
        }
      />
      <CountryListElement
        country="Test outcome label 2"
        flagUrl="/base/flags/path/2220.svg"
        onClick={[Function]}
        outcome={
          Object {
            "id": 2200,
            "label": "Test outcome label 2",
            "participantId": 2220,
          }
        }
      />
    </CountryList>
  </TabPagination>
</div>
`;
